src/commands/deploy/getDeployNode.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, Uri } from "vscode";
+import { configurationSettings, none } from "../../constants";
+import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { WebAppTreeItem } from "../../explorer/WebAppTreeItem";
+import { ext } from '../../extensionVariables';
+import { getWorkspaceSetting, updateWorkspaceSetting } from "../../vsCodeConfig/settings";
+import { IDeployContext, WebAppSource } from "./IDeployContext";
+
+export interface IDeployNode {
+    node: SiteTreeItem;
+    isNewWebApp: boolean;
+}
+
+export async function getDeployNode(context: IDeployContext, target: Uri | string | SiteTreeItem | undefined, confirmDeployment: boolean): Promise<IDeployNode> {
+    const defaultWebAppId: string | undefined = getWorkspaceSetting(configurationSettings.defaultWebAppToDeploy, context.workspace.uri.fsPath);
+    let node: SiteTreeItem | undefined;
+
+    // if the entry point for deploy was via "Deploy" after creating, it is also a new web app (so confirmDeployment would be false)
+    let isNewWebApp: boolean = !confirmDeployment;
```

Comment:
Can we change the signature of `deploy` to be something like this:
```typescript
export async function deploy(context: IActionContext, target?: vscode.Uri | SiteTreeItem | undefined, isTargetNewWebApp?: boolean): Promise<void> {
```
(I tried using `isNewWebApp` as the name first, but that causes a weird conflict)

For two reasons:
1. I don't like the name `confirmDeployment` in this function. Perhaps there are other reasons we let users skip confirming deployment in the future and thus it wouldn't be accurate to do `isNewWebApp: boolean = !confirmDeployment;`
1. Then you can simplify extension.ts like this:
```typescript
// before
registerCommand('appService.Deploy', async (actionContext: IActionContext, target?: vscode.Uri | WebAppTreeItem | undefined) => {
    await deploy(actionContext, true, target);
});


// after
registerCommand('appService.Deploy', deploy);
```